pulsar-common/src/main/java/com/yahoo/pulsar/common/api/Commands.java

```diff
@@ -534,15 +534,19 @@ public static ByteBuf newConsumerStatsResponse(CommandConsumerStatsResponse.Buil
     private final static ByteBuf cmdPing = serializeWithSize(
             BaseCommand.newBuilder().setType(Type.PING).setPing(CommandPing.getDefaultInstance()));
 
-    static ByteBuf newPing() {
-        return RecyclableDuplicateByteBuf.create(cmdPing).retain();
+    public static ByteBuf newPing() {
+        ByteBuf pingCmdBuf =  RecyclableDuplicateByteBuf.create(cmdPing).retain();
```

Comment:
It cannot be just `return cmdPing` because the buffer has a read pointer, once you write on the channel, the read pointer will be moved and the buffer will look empty next time.